ilastik/applets/dataSelection/dataSelectionApplet.py

```diff
@@ -105,6 +121,87 @@ def topLevelOperator(self):
+
+        arg_parser = argparse.ArgumentParser()
+        for role_name in role_names or []:
+            arg_name = cls._role_name_to_arg_name(role_name)
+            arg_parser.add_argument(
+                "--" + arg_name, "--" + arg_name.replace("_", "-"),
+                nargs="+",
+                help=f"List of input files for the {role_name} role",
+            )
+
+        # Finally, a catch-all for role 0 (if the workflow only has one role, there's no need to provide role names
+        arg_parser.add_argument(
+            "unspecified_input_files",
+            nargs="*",
+            help="List of input files to process.",
+            action=make_trailing_args_action(role_names),
+        )
+
+        arg_parser.add_argument(
+            "--preconvert-stacks", "--preconvert_stacks",
+            help="Convert image stacks to temporary hdf5 files before loading them.",
+            action="store_true",
+            default=False,
+        )
+
+        def parse_input_axes(raw_input_axes: str) -> List[Optional[vigra.AxisTags]]:
+            input_axes = [axes.strip() for axes in raw_input_axes.split(",")]
+            if len(input_axes) > len(role_names):
+                raise ValueError("Specified input axes exceed number of data roles ({role_names})")
+            if len(input_axes) == 1:
+                input_axes = input_axes * len(role_names)
+            else:
+                input_axes += ["None"] * (len(role_names) - len(input_axes))
+            return [parse_axiskeys(keys) for keys in input_axes]
+
+        arg_parser.add_argument(
+            "--input-axes", "--input_axes",
+            help=("Dataset axes names; a list of comma-separated axis names, representing how datasets are to be "
+                  "interpreted in each workflow role e.g.: 'xyz,xyz' ."
+                  " If a single value is provided, it is assumed to apply to all roles ({role_names}). If more than "
+                  "one value is provided but less than the total number of roles, the missing roles will have their "
+                  "axistags defined by the training data axistags. If --ignore-training-axistags is set, dataset axistags "
+                  "will be inferred from the file conventions, e.g.: 'axes' in .n5, and not from the "
+                  "training data axistags. Defaults to using the training data axistags"),
+            required=False,
+            type=parse_input_axes,
+            default=[None] * len(role_names)
+        )
+
+        arg_parser.add_argument(
+            "--ignore-training-axistags", "--ignore_training_axistags",
+            help="Do not use training data to guess input data axis on headless runs. Can still use file axis conventions",
+            action="store_true",
+        )
+
+        arg_parser.add_argument(
+            "--stack-along",
+            "--stack_along",
+            help="Axis along which stack datasets (e.g.: my_yx_slices*.tiff) are to be stacked",
```

Comment:
again, I would say that we should say in this doc string that the stacked axes comes last - otherwise some parameters (subregion and so on) will be impossible to infer.

Reply:
I've added some more help text to clarify what will happen with stacked images and the axis along which they stack